### PR TITLE
Produce error on prompt in non-tty environment.

### LINKED ISSUE #495: prompt() fails silently if stdin isn't a TTY
When run with `node test.js < /dev/null`, this script exits with code 0 and without logging anything:

```javascript
var inquirer = require('inquirer');

inquirer.prompt([{name: 'username', message: 'username'}])
.then(
    function (answers) {
      console.log('Answers:');
      console.log(answers);
  },
  function (err) {
    console.log('Error:');
    console.log(err.stack);
    process.exit(1);
  }
);
```

This can cause some weird problems in CI environments, where scripts succeed without doing anything.

I encountered this with inquirer 3.0.1 on Node.js 6.9.1.

### PR DESCRIPTION
**Pull request by jhorbulyk**
Fixes #495

### CODE REVIEW

File: packages/inquirer/lib/inquirer.js [hunk 1]
```diff
@@ -24,6 +24,18 @@ inquirer.ui = {
  */
 inquirer.createPromptModule = function(opt) {
   var promptModule = function(questions) {
+    // Check if prompt is being called in TTY environment
+    // If it isn't return a failed promise
+    if (!process.stdin.isTTY) {
+      return new Promise(function(resolve, reject) {
```
Comment (SBoudrias) [inline packages/inquirer/lib/inquirer.js]: Let's replace with `return Promise.reject(error)`

**Comment by SBoudrias** [comment]
The rest looks good to me, we should be able to merge and release quickly.

**Comment by jhorbulyk** [comment]
@SBoudrias Updated

**Comment by SBoudrias** [comment]
Perfect thanks! I'll make a new release soon.

**Comment by PetrSnobelt** [comment]
Hi, this MR causes the problem when used with `when` property, for example:
```
responses = await inquirer.prompt([{
          name: 'name',
          message: 'give a name to your app',
          default: '@kiwicom/fe-app',
          when: () => !flags.name,
        }])
```
We provide name in flags on CI, but it break our CI with "Error: Prompts can not be meaningfully rendered in non-TTY environments"

**Comment by jhorbulyk** [comment]
@PetrSnobelt Fixed in https://github.com/SBoudrias/Inquirer.js/pull/896

**Comment by ruyadorno** [comment]
Hi @SBoudrias and @jhorbulyk sorry but unfortunately this change introduces a regression since `stdin` can be configured to something else other than just `process.stdin`: 

https://github.com/SBoudrias/Inquirer.js/blob/2c521b70d45073a9a4d50ab3ae9481fd6104c1a4/packages/inquirer/lib/ui/baseUI.js#L70

it completely broke [ipt](https://github.com/ruyadorno/ipt) in your latest patch release since it actually leverages the configurable `stdin` in order to provide the functionality there 😭

**Comment by dan-barrett** [comment]
I'm somewhat confused as to where this issue stands. Looks like it was partly reverted in a future commit?

Can I explicitly configure inquirer to fail in non-tty environments or is that no longer possible? Having hard time finding documentation about it. Thanks!

**Comment by jhorbulyk** [comment]
> I'm somewhat confused as to where this issue stands. 

My understanding is that after the latest merged PRs, inquirer will fail in non-tty environments but only if:
* inquirer hasn't been configured to consume something other than `stdin`
* `when()` functions still cause the prompt to be rendered

**Pull request merged**